server/controllers/submission.js

```diff
@@ -16,13 +18,35 @@ exports.getAllSubmissions = asyncHandler(async (req, res, next) => {
   }
 
   try {
-    const isOwner = await Contest.fineOne({ userId: userId });
+    const isOwner = await Contest.findOne({ userId: userId });
     if (isOwner) {
-      const submissions = Submission.find({ contestId: contestId });
-      res.status(200).json({ submission: submissions });
+      const submissions = await Submission.find({ contestId: contestId });
+      submissions.map(async (submission) => {
+        const name = await User.findOne({ _id: submission.userId });
+        const submissionData = {
+          _id: submission._id,
+          name: name.username,
+          files: submission.files,
+        };
+        submissionList.push(submissionData);
+        if (submissionList.length === submissions.length) {
+          res.status(200).json({ submission: submissionList, isOwner: true });
+        }
+      });
     } else {
-      const submissions = Submission.find({ userId: userId });
-      res.status(200).json({ submission: submissions });
+      const submissions = await Submission.find({ userId: userId });
+      submissions.map(async (submission) => {
```

Comment:
Array.map returns an array, when you aren't using the returned array use forEach instead

Reply:
optimize the controller, using forEach